response_operations_ui/views/respondents.py

```diff
@@ -14,22 +18,93 @@
                           static_folder='static', template_folder='templates')
 
 
-@respondent_bp.route('/', methods=['GET', 'POST'])
+@respondent_bp.route('/', methods=['GET'])
+@login_required
+def respondent_home():
+    return render_template('respondent-search/search-respondents.html',
+                           form=RespondentSearchForm(),
+                           breadcrumbs=[{"title": "Respondents"}])
+
+
+@respondent_bp.route('/search', methods=['POST'])
+@login_required
+def search_redirect():
+    form = RespondentSearchForm()
+    form_valid = form.validate()
+
+    if not form_valid:
+        flash('At least one input should be filled')
+
+    source = form.source.data or 'home'
+    page = request.args.get('page', 1)
+
+    query_string = urlencode({
+        'email_address': form.email_address.data or '',
+        'first_name': form.first_name.data or '',
+        'last_name': form.last_name.data or '',
+        'source': source,
+        'page': page
+    })
+
+    if not form_valid:
+        redirect_url = url_for('respondent_bp.respondent_home')
+    else:
+        redirect_url = urljoin(url_for('respondent_bp.respondent_search'), '?' + query_string)
```

Comment:
url_for allows for parameters to be added.  https://stackoverflow.com/questions/7478366/create-dynamic-urls-in-flask-with-url-for 

It's better to do it all here instead of constructing parameter strings yourself (this is true in almost every instance I think, any tool that deals with urls will usually have its own inbuilt way to pass parameters).
Additionally, if the form isn't valid then it does a bunch of stuff it doesn't need to do.  The code should be something like 
```python
if not form_valid:
    #do all the invalid stuff
else:
    return redirect(url_for('respondent_bp.respondent_home'))
```

Reply:
Refactored in next commit